### PR TITLE
# 25 fix tests

### PR DESCRIPTION
**Pull request by KhangZhengNg**
### Related Issues
#25 

### Description
fix tests that ignored previously

### How to check PR
Run ScheduleReportActivityTest.java

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java]: move this into the OnStart() like  where the viewmodel if check is done. Should cause it to get some coverage

File: app/src/main/java/com/example/bigowlapp/utils/GeoLocationFormatter.java
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/utils/GeoLocationFormatter.java:12]: add empty list check in the case the location formatter gets 0 results. Right now the app crashes in that case.

File: app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java]: Flip/Invert the if statement in line 49 to reduce complexity. If you click on the if and then click lightbulb, and then click invert if  in Android Studio, it will do this automatically

Also if you have the time, you can add a test case that covers the `schedule = null` scenario

https://github.com/Landry333/Big-Owl/blob/b973729ce9a0ef0039a049b96c5be7683728a256/app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java#L50-L54

**Review by refatK — CHANGES_REQUESTED** [review]
Just requesting some quality things, but good job getting the tests to work.

**Review by refatK — APPROVED** [review]
Seems good now.

Edit: Oops one more thing.

File: app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/ScheduleReportActivity.java]: you can remove this now. Since its called in OnStart. After that, I'll approve

**Review by refatK — DISMISSED** [review]

**Review by ShawnGuerra-Bautista — APPROVED** [review]
Tests work perfectly.

Approved.

**Review by refatK — COMMENTED** [review]
Thanks for addressing the request change

Since the quality thing was added, I will approve once all sonar errors are resolved

**Review by refatK — APPROVED** [review]
Its good now.
Zero Sonar and lint errors.

so approved!

**Pull request merged**
